src/streams/streamsActions.js

```diff
@@ -26,21 +27,20 @@ export const updateExistingStream = (
   initialValues: Object,
   newValues: Object,
 ) => async (dispatch: Dispatch, getState: GetState) => {
+  const apiParams: UpdateStreamApiParamsType = {};
```

Comment:
This looks like a candidate for a utility function.
The chances are that we will add more fields and we will need this logic in other places.
What if you do create a `currentValues` object with the values needed. 
Then we can diff it against the current version (via a utility function)

Reply:
Then we need to do some refactors, like
* currently we have `name` key (for stream name) in `initialValues` and `newValues`
But at the time of sending parameters to API, we want key `new_name`, so again a if/else will be required.
Suggestions?

Reply:
I would suggest you do an object diffing function and then pass the resulting object (if it is not empty!) to this new function:
https://github.com/zulip/zulip-mobile/blob/master/src/api/apiFetch.js#L14